src/services/eksCluster/data.ts

```diff
@@ -123,9 +125,61 @@ const getResourceTags = async (eks: EKS, arn: string): Promise<TagMap> =>
     }
   })
 
+const listNodegroups = async ({
+  eks,
+  clusterName,
+}: {
+  eks: EKS
+  clusterName: string
+}): Promise<string[]> =>
+  new Promise<string[]>(resolve => {
+    const nodeGroupsList: string[] = []
+    let args: ListNodegroupsRequest = { clusterName }
+    const listAllNodeGroups = (token?: string): void => {
+      if (token) {
+        args = { ...args, nextToken: token }
+      }
+
+      try {
+        eks.listNodegroups(
+          args,
+          (err: AWSError, data: ListNodegroupsResponse) => {
+            if (err) {
+              errorLog.generateAwsErrorLog({
+                functionName: 'eks:listNodegroups',
+                err,
+              })
+            }
+
+            /**
+             * No node groups for this region
+             */
+            if (isEmpty(data)) {
+              return resolve(nodeGroupsList)
+            }
+
+            const { nextToken, nodegroups } = data || {}
+
+            nodeGroupsList.push(...nodegroups)
+
+            if (nextToken) {
+              listAllNodeGroups(nextToken)
+            }
+
+            resolve(nodeGroupsList)
```

Comment:
missing else... the promise will be immediately resolved while it continues execution